src/UI/Settings/ProviderSettingsModelBase.js

```diff
@@ -3,6 +3,66 @@ var DeepModel = require('backbone.deepmodel');
 var Messenger = require('../Shared/Messenger');
 
 module.exports = DeepModel.DeepModel.extend({
+    connectData : function(action) {
+        var self = this;
+        
+        this.trigger('connect:sync');
+
+        var promise = $.Deferred();
```

Comment:
Yea, but i didn't want the promise to resolve until all the sub parts were done. For OAuth, you need to do a couple client/server calls.